sdk/python/cmd/pulumi-language-python/main.go

```diff
@@ -960,5 +959,62 @@ func (host *pythonLanguageHost) GetProgramDependencies(
 func (host *pythonLanguageHost) RunPlugin(
 	req *pulumirpc.RunPluginRequest, server pulumirpc.LanguageRuntime_RunPluginServer,
 ) error {
-	return errors.New("not supported")
+	logging.V(5).Infof("Attempting to run python plugin in %s", req.Program)
+
+	args := []string{req.Program}
+	args = append(args, req.Args...)
+
+	var cmd *exec.Cmd
+	var err error
+	var virtualenv string
+	if host.virtualenv != "" {
+		virtualenv = host.virtualenvPath
+		if !python.IsVirtualEnv(virtualenv) {
+			return python.NewVirtualEnvError(host.virtualenv, virtualenv)
+		}
+		cmd = python.VirtualEnvCommand(virtualenv, "python", args...)
+	} else {
+		cmd, err = python.Command(server.Context(), args...)
```

Comment:
optional: err is needed only for the else block. Code down below gets its own err.
Can move the declaration closer to where it's used, reducing its scope.
```suggestion
	var virtualenv string
	if host.virtualenv != "" {
		virtualenv = host.virtualenvPath
		if !python.IsVirtualEnv(virtualenv) {
			return python.NewVirtualEnvError(host.virtualenv, virtualenv)
		}
		cmd = python.VirtualEnvCommand(virtualenv, "python", args...)
	} else {
		var err error
		cmd, err = python.Command(server.Context(), args...)
```

That makes the reading flow of this code block nicer.
Like, `var X, Y; if { ... } else { ... }` suggests that X and Y are outputs of that "paragraph" of code.
In this case, only cmd and virtualenv are outputs of this block.